salt/runners/pillar.py

```diff
@@ -82,16 +83,18 @@ def show_pillar(minion='*', **kwargs):
         pillar = runner.cmd('pillar.show_pillar', [])
         print(pillar)
     '''
+    # Don't stomp on the master opts
+    opts = copy.deepcopy(__opts__)
 
     saltenv = 'base'
-    pillarenv = __opts__['pillarenv'] if 'pillarenv' in __opts__ else None
-    id_, grains, _ = salt.utils.minions.get_minion_data(minion, __opts__)
+    pillarenv = opts['pillarenv'] if 'pillarenv' in opts else None
+    id_, grains, _ = salt.utils.minions.get_minion_data(minion, opts)
     if grains is None:
         grains = {'fqdn': minion}
 
     for key in kwargs:
         if key == 'pillarenv':
-            __opts__['pillarenv'] = kwargs[key]
+            opts['pillarenv'] = kwargs[key]
```

Comment:
About this...this was introduced in https://github.com/ogd-software/salt/commit/8d19790cdab909042b8ded1b5321e8f7e9f4945d but the check for `key == 'pillarenv'` now happens twice...one to modify opts and one to set the `pillarenv` variable which is also passed into `salt.pillar.Pillar`. Why is it neccesary to modify `opts['pillarenv']`  here (in favour of the `pillarenv` variable which is also passed)?

Reply:
Yeah, you're right... I didn't give this a close enough look. This is fixed in https://github.com/saltstack/salt/pull/46145/commits/2a185855ea89ac5bd9a87a80914cfb5a81d9cf7e

Reply:
Also added 467ff841 (I had forgotten to save before committing and still had one minor change that should have been in 2a18585).